certora/specs/ghoVariableDebtToken-rayMulDiv-summarization.spec

```diff
@@ -0,0 +1,276 @@
+    function mint(address, address, uint256, uint256) external returns (bool, uint256);
+    function burn(address ,uint256 ,uint256) external returns (uint256);
+    function scaledBalanceOf(address) external returns (uint256) envfree;
+    function getBalanceFromInterest(address) external returns (uint256) envfree;
+    function rebalanceUserDiscountPercent(address) external;
+    function updateDiscountDistribution(address ,address ,uint256 ,uint256 ,uint256) external;
+}
+
+ghost rayMul_gst(mathint , mathint) returns uint256 {
+    //axiom 1==1;
+        axiom forall mathint x. forall mathint y. //rayMul_gst(x,y)+0 == x;
+      (
+       ((x==0||y==0) => rayMul_gst(x,y)==0)
+       &&
+       x <= to_mathint(rayMul_gst(x,y)) && to_mathint(rayMul_gst(x,y)) <= 2*x
+      )    ;
+}
+ghost rayDiv_gst(mathint , mathint) returns uint256 {
+    //    axiom 1==1;
+        axiom forall mathint x. forall mathint y. //rayDiv_gst(x,y)+0 == x;
+      (
+       x/2 <= to_mathint(rayDiv_gst(x,y)) && to_mathint(rayDiv_gst(x,y)) <= x
+      );
+}
+
+
+definition MAX_DISCOUNT() returns uint256 = 10000; // equals to 100% discount, in points
+
+ghost mapping(address => mapping (uint256 => uint256)) discount_ghost;
+ghost mapping(uint256 => uint256) index_ghost;
+
+/**
+* Query index_ghost for the index value at the input timestamp
+**/
+function indexAtTimestamp(uint256 timestamp) returns uint256 {
+    require index_ghost[timestamp] >= ray();
+    return index_ghost[timestamp];
+}
+
+/**
+* Query discount_ghost for the [user]'s balance of discount token at [timestamp]
+**/
+function balanceOfDiscountTokenAtTimestamp(address user, uint256 timestamp) returns uint256 {
+	return discount_ghost[user][timestamp];
+}
+
+
+function get_discount_scaled(env e, address user) returns uint256 {
+    uint256 user_scaledBal_prev = scaledBalanceOf(user);
+    
+    //    assert (user_scaledBal_after <= user_scaledBal_prev);
+    uint256 current_index = indexAtTimestamp(e.block.timestamp);
+    uint256 user_index = getUserCurrentIndex(user);
+    require user_index <= current_index;
+
+    //uint256 bal_increase = (current_index-sender_index) * previousScaledBalance_of_sender;
+    mathint bal_increase = rayMul_gst(user_scaledBal_prev, current_index) -
+        rayMul_gst(user_scaledBal_prev, user_index);
+
+    //uint256 discountScaled = bal_increase * sender_precentage / index;
+    uint256 discountPercent = getDiscountPercent(e, user);
+    uint256 discount = require_uint256(bal_increase * discountPercent / MAX_DISCOUNT());
+    uint256 discountScaled = rayDiv_gst(discount, current_index);
+
+    return discountScaled;
+}
+
+
+
+/*================================================================================
+  Calling to mint(...amount) can't increase the balance by more than sclaed-amount.
+  (This catches mutant 11)
+  =================================================================================*/
+rule mint_cant_increase_bal_by_more_than_amountScaled() {
```

Comment:
this looks exactly like the rule at the bottom called mutant 11. See the comments there and remove one of the

Reply:
Correct. I commited the file ghoVariableDebtToken.sol by mistake.